x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationIssue.java

```diff
@@ -175,4 +190,233 @@ public int hashCode() {
+        public Meta getIntersection(Meta another) {
+            final List<Action> actionsIntersection;
+            if (actions != null && another.actions != null) {
+                List<Action> combinedActions = this.actions.stream()
+                    .filter(action -> action instanceof RemovalAction == false)
+                    .collect(Collectors.toList());
+                Optional<Action> thisRemovalAction = this.actions.stream().filter(action -> action instanceof RemovalAction).findFirst();
+                Optional<Action> otherRemovalAction = another.actions.stream()
+                    .filter(action -> action instanceof RemovalAction)
+                    .findFirst();
+                if (thisRemovalAction.isPresent() && otherRemovalAction.isPresent()) {
+                    Optional<List<String>> removableSettingsOptional = ((RemovalAction) thisRemovalAction.get()).getRemovableSettings();
+                    List<String> removableSettings = removableSettingsOptional.map(
+                        settings -> settings.stream()
+                            .distinct()
+                            .filter(
+                                setting -> ((RemovalAction) otherRemovalAction.get()).getRemovableSettings()
+                                    .map(list -> list.contains(setting))
+                                    .orElse(false)
+                            )
+                            .collect(Collectors.toList())
+                    ).orElse(Collections.emptyList());
+                    if (removableSettings.isEmpty() == false) {
+                        combinedActions.add(new RemovalAction(removableSettings));
+                    }
+                }
+                actionsIntersection = combinedActions;
+            } else {
+                actionsIntersection = null;
+            }
+            return new Meta(actionsIntersection, nonActionMetadata);
+        }
+
+        /*
+         * Returns an Optional Meta object from a DeprecationIssue's meta Map. If the meta Map is null then the Optional will not be
+         * present.
+         */
+        @SuppressWarnings("unchecked")
+        public static Optional<Meta> fromMetaMap(Map<String, Object> metaMap) {
+            if (metaMap == null) {
+                return Optional.empty();
+            }
+            List<Map<String, Object>> actionMaps = (List<Map<String, Object>>) metaMap.get(ACTIONS_META_FIELD);
+            List<Action> actions;
+            if (actionMaps == null) {
+                actions = null;
+            } else {
+                actions = new ArrayList<>();
+                for (Map<String, Object> actionMap : actionMaps) {
+                    final Action action;
+                    if (REMOVE_SETTINGS_ACTION_TYPE.equals(actionMap.get(ACTION_TYPE))) {
+                        action = RemovalAction.fromActionMap(actionMap);
+                    } else {
+                        action = UnknownAction.fromActionMap(actionMap);
```

Comment:
Do we need the `UnknownAction` at all, or can we be strict and throw an exception if it's not in the format we expect? Do you think that would be too strict here? (especially since this `_meta` map the user has no control over, so I think we'd only be limiting plugins)

Reply:
Yeah I was afraid I'd break some plugin that I don't know about. If you're certain that no one else is putting in any other `action`s, I can remove this.